srml/balances/src/lib.rs

```diff
@@ -592,26 +689,33 @@ where
 	}
 }
 
-impl<T: Trait<I>, I: Instance> TransferAsset<T::AccountId> for Module<T, I> {
-	type Amount = T::Balance;
+/// Simple implementation of OnUnbalancedDecrease/OnUnbalancedIncrease that
+/// just burns and mints tokens as necessary.
+pub struct BurnAndMint<T: Trait>(rstd::marker::PhantomData<T>);
 
-	fn transfer(from: &T::AccountId, to: &T::AccountId, amount: T::Balance) -> Result {
-		Self::make_transfer(from, to, amount)
+impl<T: Trait> OnUnbalancedDecrease<T::Balance> for BurnAndMint<T> {
+	fn on_unbalanced_decrease(amount: T::Balance) -> Result {
+		<Module<T>>::decrease_total_issuance_by(amount)
 	}
+}
 
-	fn withdraw(who: &T::AccountId, value: T::Balance, reason: WithdrawReason) -> Result {
-		let b = Self::free_balance(who);
-		ensure!(b >= value, "account has too few funds");
-		let new_balance = b - value;
-		Self::ensure_account_can_withdraw(who, value, reason, new_balance)?;
-		Self::set_free_balance(who, new_balance);
-		Self::decrease_total_stake_by(value);
-		Ok(())
+impl<T: Trait> OnUnbalancedIncrease<T::Balance> for BurnAndMint<T> {
+	fn on_unbalanced_increase(amount: T::Balance) -> Result {
+		<Module<T>>::increase_total_issuance_by(amount)
 	}
+}
 
-	fn deposit(who: &T::AccountId, value: T::Balance) -> Result {
-		Self::set_free_balance_creating(who, Self::free_balance(who) + value);
-		Self::increase_total_stake_by(value);
+impl<T: Trait> MakePayment<T::AccountId> for Module<T> {
+	fn make_payment(transactor: &T::AccountId, encoded_len: usize) -> Result {
+		let b = Self::free_balance(transactor);
+		let transaction_fee = Self::transaction_base_fee() + Self::transaction_byte_fee() * <T::Balance as As<u64>>::sa(encoded_len as u64);
+		if b < transaction_fee + Self::existential_deposit() {
+			return Err("not enough funds for transaction fee");
```

Comment:
Why is it not possible to go below `existential_deposit`? We don't want the account to be killed by paying fees?

Reply:
Correct.

Reply:
We assume that the account exists at the point of dispatch.